src/cluster/cluster.cc

```diff
@@ -93,32 +83,27 @@ Status Cluster::SetNodeId(const std::string &node_id) {
   return Status::OK();
 }
 
-// Set the slot to the node if new version is current version +1. It is useful
-// when we scale cluster avoid too many big messages, since we only update one
-// slot distribution and there are 16384 slot in our design.
-//
+// Set the slots to the node if new version is current version +1. It is useful
+// when we scale cluster avoid too many big messages. By th way,there are 16384 slots
+// in our design.
 // The reason why the new version MUST be +1 of current version is that,
 // the command changes topology based on specific topology (also means specific
 // version), we must guarantee current topology is exactly expected, otherwise,
 // this update may make topology corrupt, so base topology version is very important.
 // This is different with CLUSTERX SETNODES commands because it uses new version
 // topology to cover current version, it allows kvrocks nodes lost some topology
 // updates since of network failure, it is state instead of operation.
-Status Cluster::SetSlot(int slot, const std::string &node_id, int64_t new_version) {
-  // Parameters check
+Status Cluster::SetSlotRanges(const std::vector<std::pair<int, int>> &slot_ranges, const std::string &node_id,
```

Comment:
Personally, I think it's better to use alias declaration(Like `using SlotRange = std::pair<int, int>` ) or struct, because I see `std::pair<int, int>` appear 6 times.

Reply:
OK, I use SlotRange instead of std::pair<int,int>.
And it lefts the `start` and `end` integers in SlotInfo struct. Maybe we think about it later. 

```cpp
// TODO: replace integer `start`, `end` with SlotRange
struct SlotInfo {
  int start;
  int end;
  struct NodeInfo {
    std::string host;
    int port;
    std::string id;
  };
  std::vector<NodeInfo> nodes;
};
```